x/superfluid/keeper/migrate.go

```diff
@@ -10,102 +10,102 @@ import (
+		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, 0, 0, err
 	}
 
-	gammSharesInLock := preMigrationLock.Coins[0]
-
 	// Superfluid undelegate the superfluid delegated position.
-	// This deletes the connection between the lock and the intermediate account, deletes the synthetic lock, and burns the synthetic osmo.
+	// This deletes the connection between the gamm lock and the intermediate account, deletes the synthetic lock, and burns the synthetic osmo.
 	intermediateAccount, err := k.SuperfluidUndelegateToConcentratedPosition(ctx, sender.String(), lockId)
 	if err != nil {
-		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, err
+		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, 0, 0, err
 	}
 
 	// Force unlock, validate the provided sharesToMigrate, and exit the balancer pool.
 	// This will return the coins that will be used to create the concentrated liquidity position.
-	exitCoins, err := k.validateSharesToMigrateUnlockAndExitBalancerPool(ctx, sender, poolIdLeaving, preMigrationLock, sharesToMigrate, tokenOutMins)
+	// It also returns the lock object that contains the remaining shares that were not used in this migration.
+	exitCoins, remainingSharesLock, err := k.validateSharesToMigrateUnlockAndExitBalancerPool(ctx, sender, poolIdLeaving, preMigrationLock, sharesToMigrate, tokenOutMins, remainingLockTime)
 	if err != nil {
-		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, err
+		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, 0, 0, err
 	}
 
 	// Create a full range (min to max tick) concentrated liquidity position, lock it, and superfluid delegate it.
 	positionId, amount0, amount1, liquidity, joinTime, concentratedLockId, err = k.clk.CreateFullRangePositionLocked(ctx, poolIdEntering, sender, exitCoins, remainingLockTime)
 	if err != nil {
-		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, err
+		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, 0, 0, err
 	}
 	err = k.SuperfluidDelegate(ctx, sender.String(), concentratedLockId, intermediateAccount.ValAddr)
 	if err != nil {
-		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, err
+		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, 0, 0, 0, 0, err
 	}
 
-	// If there are remaining gamm shares, we must re-lock them.
-	remainingGammShares := gammSharesInLock.Sub(sharesToMigrate)
-	if !remainingGammShares.IsZero() {
-		// Create a new lock with the remaining gamm shares for the remaining lock time.
-		newLock, err := k.lk.CreateLock(ctx, sender, sdk.NewCoins(remainingGammShares), remainingLockTime)
-		gammLockId = newLock.ID
+	// If there are any remaining gamm shares after the migration, we must re-superfluid delegate them as they were previously in the gamm pool.
+	if remainingSharesLock.ID != 0 {
+		gammLockId = remainingSharesLock.ID
+		// Superfluid delegate the gamm lock.
+		err = k.SuperfluidDelegate(ctx, sender.String(), remainingSharesLock.ID, valAddr)
```

Comment:
nit: made me wonder why we do the assignment but still use `remainingSharesLock.ID`
```suggestion
		gammLockId = remainingSharesLock.ID
		// Superfluid delegate the gamm lock.
		err = k.SuperfluidDelegate(ctx, sender.String(), gammLockId, valAddr)
```

Reply:
This no longer exists since we implemented partial superfluid undelegation :)